test/rules/unnecessary_this.dart

```diff
@@ -4,6 +4,32 @@
 
 // test w/ `pub run test -N unnecessary_this`
 
+void bar() {
+
+}
+
+int x;
+
+
+class Parent {
+  int x, y;
+  void bar() {
+
+  }
+}
+
+class Child extends Parent {
+  void foo() {
+    this.bar(); // LINT
```

Comment:
This is wrong. Without the `this`, `bar()` will refer to the top-level function, which is clearly a change is semantics.